staging/src/k8s.io/apimachinery/pkg/runtime/converter.go

```diff
@@ -460,7 +459,7 @@ func DeepCopyJSONValue(x interface{}) interface{} {
 			clone[i] = DeepCopyJSONValue(v)
 		}
 		return clone
-	case string, int64, bool, float64, nil, encodingjson.Number:
+	case string, int, int32, int64, bool, float64, nil, encodingjson.Number:
```

Comment:
These and other (like `int16` and `byte`) types are missing by design - this is how stdlib deserializes JSON into numbers.

Reply:
that's interesting, I wasn't aware of that :)